docs/theories/strings.rst

```diff
@@ -0,0 +1,66 @@
+Theory Reference: Strings
+=========================
+
+cvc5 supports all operators of the `SMT-LIB standard for strings
+<https://smtlib.cs.uiowa.edu/theories-UnicodeStrings.shtml>`_. It additionally
+supports some non-standard operators that are described below.
+
+Semantics
+^^^^^^^^^
+
+.. code-block::
+
+   * (str.indexof_re String RegLan Int Int)
+
```

Comment:
Why are there 3 arguments?  It looks like the third argument is not used in the definition.

Reply:
Good catch. I have updated the definition.